### PR TITLE
Dịch "linear-regression-scratch_vn" - Phần 6

### LINKED ISSUE #622: Dịch "linear-regression-scratch_vn" - Phần 6
Phần cần dịch: https://github.com/aivivn/d2l-vn/blame/master/chapter_linear-networks/linear-regression-scratch_vn.md#L254-L296

### PR DESCRIPTION
**Pull request by quangnhat185**
## Pull Request Checklist

**Đánh dấu tick (sau khi "Create Pull Request") vào các mục dưới đây:**

* [ ] Pull Request này tương ứng với issue nào? Trả lời số issue sau dấu `#` (không có dấu cách): Close #622 

* [x] Bản dịch này có bám sát **[Bảng thuật ngữ](https://github.com/aivivn/d2l-vn/blob/master/glossary.md)** không? Nếu một từ/cụm chưa có trong Bảng thuật ngữ, hãy tạo một Pull Request riêng đề xuất cách dịch từ/cụm đó.

* [x] Format của bản gốc có được giữ nguyên không?
    * Các phần in nghiêng, in đậm.
    * Tiêu đề (số lượng dấu `#` đầu dòng).
    * Bảng biểu, chú thích cho bảng (dòng phía trên mỗi bảng bắt đầu bằng dấu `:`).
    * Dịch các từ trong hình vẽ nếu cần. Các dòng có hình có dạng: `caption`.
    * Dịch các chú thích hình vẽ (thay các cụm `*dịch chú thích ảnh phía trên*` bằng bản dịch tiếng Việt).
    * Không dịch các phần code (nằm giữa hai dấu `````).
    * Copy các công thức toán từ bản gốc (các đoạn có `$`).
    * Giữ các dòng gán nhãn (bắt đầu với `:label:`).
    * Không tự thêm bớt các dòng trắng.
     
* [x] Trong một đoạn văn, mỗi câu văn đã được viết trong một dòng, giữa các dòng không có dòng trắng. Markdown sẽ coi những dòng liền nhau không có dòng trắng là một đoạn văn. Việc này giúp công đoạn review được thuận tiện hơn.

* [x] Tất cả chữ trong các hình vẽ đã được dịch. Nếu chưa, vui lòng xem thêm **[hướng dẫn hoặc yêu cầu dịch hình](https://github.com/aivivn/d2l-vn/blob/master/img/README.md)**.

* [x] Điền tên của bạn vào mục "Những người thực hiện" ở cuối file.

* [ ] **Sau khi được approve**, điền tên những reviewers mà bạn thấy có nhiều đóng góp cho Pull Request này. Xem danh sách Contributors **[tại đây](https://github.com/aivivn/d2l-vn/blob/master/docs/contributors_info.md)**.

## Với Reviewer

1. Chỉ review nếu Pull Request này đã sẵn sàng cho việc review.

2. **Bạn có thể đề xuất thay đổi cách dịch mỗi dòng trực tiếp như trong hình dưới đây:**
img

### CODE REVIEW

File: chapter_linear-networks/linear-regression-scratch_vn.md [hunk 1]
```diff
@@ -284,7 +284,10 @@ In the implementation, we need to transform the true value `y` into the predicte
 The result returned by the following function will also be the same as the `y_hat` shape.
 -->
 
-*dịch đoạn phía trên*
+Bởi việc cập nhật mô hình đòi hỏi trích xuất gradient của hàm mất mát, ta cần xác định hàm mất mát trước tiên.
```
Comment (tanduong) [inline chapter_linear-networks/linear-regression-scratch_vn.md]:
```suggestion
Để cập nhật mô hình ta cần tính gradient của hàm mất mát, vậy nên ta phải định nghĩa hàm mất mát trước tiên.
```

File: chapter_linear-networks/linear-regression-scratch_vn.md [hunk 2]
```diff
@@ -284,7 +284,10 @@ In the implementation, we need to transform the true value `y` into the predicte
...
+ở đây chúng ta sẽ sử dụng hàm mất mát bình phương đã được trình bày ở phần trước.
```
Comment (tanduong) [inline chapter_linear-networks/linear-regression-scratch_vn.md]:
```suggestion
chúng ta sẽ sử dụng hàm mất mát bình phương (SE) như trình bày ở phần trước.
```

File: chapter_linear-networks/linear-regression-scratch_vn.md [hunk 3]
```diff
@@ -284,7 +284,10 @@ In the implementation, we need to transform the true value `y` into the predicte
...
+Trong quá trình thực thi, chúng ta cần chuyển đổi giá trị thực `y` sang kích thước của giá trị dự đoán `y_hat`.
```
Comment (tanduong) [inline chapter_linear-networks/linear-regression-scratch_vn.md]:
```suggestion
Trong quá trình hiện thực, chúng ta cần chuyển đổi giá trị nhãn đúng `y` sang kích thước của giá trị dự đoán `y_hat`.
```

Comment (tanduong) [reply chapter_linear-networks/linear-regression-scratch_vn.md]: Mình vẫn chưa hài lòng về `kích thước của giá trị dự đoán`. Đọc vẫn không hiểu.

File: chapter_linear-networks/linear-regression-scratch_vn.md [hunk 4]
```diff
@@ -296,7 +299,7 @@ def squared_loss(y_hat, y):
 ## Defining the Optimization Algorithm
 -->
 
-## *dịch tiêu đề phía trên*
+## Xác định Thuật toán tối ưu
```
Comment (tanduong) [inline chapter_linear-networks/linear-regression-scratch_vn.md]:
```suggestion
## Định nghĩa thuật toán tối ưu
```

Nhiều bạn dịch `Definine` là xác định lắm luôn. Này dịch là `định nghĩa` nhe!

File: chapter_linear-networks/linear-regression-scratch_vn.md [hunk 5]
```diff
@@ -305,7 +308,10 @@ Since none of the other models that this book introduces
 can be solved analytically, we will take this opportunity to introduce your first working example of stochastic gradient descent (SGD).
 -->
 
-*dịch đoạn phía trên*
+Như đã thảo luận ở mục trước, hồi quy tuyến tính có một giải pháp dạng đóng.
```
Comment (tanduong) [inline chapter_linear-networks/linear-regression-scratch_vn.md]:
```suggestion
Như đã thảo luận ở mục trước, hồi quy tuyến tính có một nghiệm (dạng đóng)[https://vi.wikipedia.org/wiki/Bi%E1%BB%83u_th%E1%BB%A9c_d%E1%BA%A1ng_%C4%91%C3%B3ng]. 
```

`solution` là nghiệm nhe!!!

File: chapter_linear-networks/linear-regression-scratch_vn.md (hunk 4 above)
Comment (quangnhat185) [reply chapter_linear-networks/linear-regression-scratch_vn.md]: Như mình đã nói ở PR trước, vì mọi người đã dịch Defining là Xác định, thế thống nhất mình giữ nguyên cách dịch này. Nếu bạn muốn thay đổi có thể mở một PR riêng để mọi người bàn luận

File: chapter_linear-networks/linear-regression-scratch_vn.md (hunk 3 above)
Comment (quangnhat185) [reply chapter_linear-networks/linear-regression-scratch_vn.md]: Mình kiến nghị thay `quá trình hiện thực` thành `thực tế`
```suggestion
Trong thực tế, chúng ta cần chuyển đổi giá trị nhãn đúng `y` sang kích thước của giá trị dự đoán `y_hat`.
```

File: chapter_linear-networks/linear-regression-scratch_vn.md [hunk 6]
```diff
@@ -284,7 +284,10 @@ In the implementation, we need to transform the true value `y` into the predicte
 The result returned by the following function will also be the same as the `y_hat` shape.
 -->
 
-*dịch đoạn phía trên*
+Để cập nhật mô hình ta cần tính gradient của hàm mất mát, vậy nên ta phải định nghĩa hàm mất mát trước tiên.
```
Comment (rootonchair) [inline chapter_linear-networks/linear-regression-scratch_vn.md]:
```suggestion
Để cập nhật mô hình ta cần tính gradient của hàm mất mát, vậy nên ta phải xác định hàm mất mát trước tiên.
```

File: chapter_linear-networks/linear-regression-scratch_vn.md [hunk 7]
```diff
@@ -284,7 +284,10 @@ In the implementation, we need to transform the true value `y` into the predicte
...
+chúng ta sẽ sử dụng hàm mất mát bình phương (SE) như trình bày ở phần trước.
```
Comment (rootonchair) [inline chapter_linear-networks/linear-regression-scratch_vn.md]:
```suggestion
Chúng ta sẽ sử dụng hàm mất mát bình phương (SE) như đã trình bày ở phần trước.
```

File: chapter_linear-networks/linear-regression-scratch_vn.md [hunk 8]
```diff
@@ -305,7 +308,10 @@ Since none of the other models that this book introduces
 can be solved analytically, we will take this opportunity to introduce your first working example of stochastic gradient descent (SGD).
 -->
 
-*dịch đoạn phía trên*
+Như đã thảo luận ở mục trước, hồi quy tuyến tính có một nghiệm (dạng đóng)[https://vi.wikipedia.org/wiki/Bi%E1%BB%83u_th%E1%BB%A9c_d%E1%BA%A1ng_%C4%91%C3%B3ng]. 
+Tuy nhiên, đây không phải là một cuốn sách về hồi quy tuyến tính, mà là cuốn sách về học sâu. 
+Vì không một mô hình nào được trình bày trong cuốn sách này
```
Comment (rootonchair) [inline chapter_linear-networks/linear-regression-scratch_vn.md]:
```suggestion
Vì không một mô hình nào khác được trình bày trong cuốn sách này 
```

File: chapter_linear-networks/linear-regression-scratch_vn.md [hunk 9]
```diff
@@ -305,7 +308,10 @@ Since none of the other models that this book introduces
 can be solved analytically, we will take this opportunity to introduce your first working example of stochastic gradient descent (SGD).
 -->
 
-*dịch đoạn phía trên*
+Như đã thảo luận ở mục trước, hồi quy tuyến tính có một nghiệm (dạng đóng)[https://vi.wikipedia.org/wiki/Bi%E1%BB%83u_th%E1%BB%A9c_d%E1%BA%A1ng_%C4%91%C3%B3ng]. 
+Tuy nhiên, đây không phải là một cuốn sách về hồi quy tuyến tính, mà là cuốn sách về học sâu. 
+Vì không một mô hình nào được trình bày trong cuốn sách này 
+có thể giải được bằng phương pháp phân tích, chúng tôi sẽ nhân đó giới thiệu với các bạn ví dụ đầu tiên về hạ gradient ngẫu nhiên (_stochastic gradient descent-SGD_)
```
Comment (rootonchair) [inline chapter_linear-networks/linear-regression-scratch_vn.md]:
```suggestion
có thể giải được bằng phương pháp phân tích, chúng tôi sẽ nhân đó giới thiệu với các bạn ví dụ đầu tiên về hạ gradient ngẫu nhiên (_stochastic gradient descent -- SGD_)
```

File: chapter_linear-networks/linear-regression-scratch_vn.md [hunk 10]
```diff
@@ -275,7 +275,7 @@ def linreg(X, w, b):
 ## Defining the Loss Function
 -->
 
-## *dịch tiêu đề phía trên*
+## Xác định Hàm mất mát
```
Comment (tanduong) [inline chapter_linear-networks/linear-regression-scratch_vn.md]: Trong đa số trường hợp thì `Define` dịch là `định nghĩa` sẽ sát hơn `xác định`. Mình bịa ra luôn chứ hông có liệt kê ra nhiều lựa chọn rồi chọn một cái.

cc @tiepvupsu

Comment (tiepvupsu) [reply chapter_linear-networks/linear-regression-scratch_vn.md]: Đúng rồi, cái này là `định nghĩa`  mà, `determine` mới là `xác định`.

Nhân tiện, bác @tanduong có nhiều comment đầy đủ quá, mời bác tham gia [slack của nhóm dịch](https://docs.google.com/forms/d/e/1FAIpQLScYforPRBn0oDhqSV_zTpzkxCAf0F7Cke13QS2tqXrJ8LxisQ/viewform) để tiện trao đổi thêm.

Comment (tiepvupsu) [inline chapter_linear-networks/linear-regression-scratch_vn.md]:
```suggestion
## Định nghĩa Hàm Mất mát
```

File: chapter_linear-networks/linear-regression-scratch_vn.md (hunk 6 above)
Comment (tiepvupsu) [reply chapter_linear-networks/linear-regression-scratch_vn.md]: `define` là `định nghĩa` mà @rootonchair.

File: chapter_linear-networks/linear-regression-scratch_vn.md (hunk 4 above)
Comment (tiepvupsu) [reply chapter_linear-networks/linear-regression-scratch_vn.md]: Hmm, anh không nhớ là mọi người thống nhất như vậy ở đâu. Em dẫn lại link.

`determine` (verb) hay `deterministic` (adj) mới là `xác định`. Ở đây đúng là `Định nghĩa` mà.

File: chapter_linear-networks/linear-regression-scratch_vn.md (hunk 10 above)
Comment (rootonchair) [reply chapter_linear-networks/linear-regression-scratch_vn.md]: Nếu nói trong toán thì từ `define` được dịch là `xác định` vẫn ổn. Trong mục này nói về cách kiếm hàm mất mát, tức là có nhiều hàm mất mát khác nhau và ta cần xác định trong bài toán này cần hàm mất mát như thế nào. Còn `định nghĩa` thường dùng để chỉ fact, lý thuyết. "Định nghĩa hàm mất mát" theo em cũng không phản ánh chuẩn nội dung trong mục.

Comment (tanduong) [reply chapter_linear-networks/linear-regression-scratch_vn.md]: Chỗ này thật ra mình vẫn có chút bâng khuâng. Nhưng mình thấy ở mức độ của phần này của sách thì đang áp đặt hơn là thảo luận.

Comment (tiepvupsu) [reply chapter_linear-networks/linear-regression-scratch_vn.md]: > Nhưng mình thấy ở mức độ của phần này của sách thì đang áp đặt hơn là thảo luận.

Bạn giải thích rõ ý câu này được không? Ý bạn là tác giả đang áp đặt hay các reviewer đang áp đặt cách dịch? Mình có thể đọc lại cả mục rồi ra quyết định.

Comment (tiepvupsu) [reply chapter_linear-networks/linear-regression-scratch_vn.md]: Tham khảo thêm 
https://vi.wikipedia.org/wiki/L%E1%BA%ADp_tr%C3%ACnh_h%C6%B0%E1%BB%9Bng_%C4%91%E1%BB%91i_t%C6%B0%E1%BB%A3ng

các bạn sẽ thấy `định nghĩa lớp`, `định nghĩa đối tượng`.

File: chapter_linear-networks/linear-regression-scratch_vn.md (hunk 4 above)
Comment (tiepvupsu) [reply chapter_linear-networks/linear-regression-scratch_vn.md]: > Như mình đã nói ở PR trước, vì mọi người đã dịch `Defining` là `Xác định` từ đầu nên thống nhất giữ nguyên cách dịch này. Nếu bạn muốn thay đổi có thể mở một PR riêng để mọi người bàn luận

Anh nghĩ mình phải quay lại sửa luôn. Anh hoàn toàn đồng ý với việc dịch là "định nghĩa". Cái này là định nghĩa hàm khi lập trình, không phải định nghĩa toán học.

File: chapter_linear-networks/linear-regression-scratch_vn.md (hunk 10 above)
Comment (tanduong) [reply chapter_linear-networks/linear-regression-scratch_vn.md]: > > Nhưng mình thấy ở mức độ của phần này của sách thì đang áp đặt hơn là thảo luận.
> 
> Bạn giải thích rõ ý câu này được không? Ý bạn là tác giả đang áp đặt hay các reviewer đang áp đặt cách dịch? Mình có thể đọc lại cả mục rồi ra quyết định.

Ý mình là tác giả nêu ra định nghĩa chứ tác giả hông thảo luận nhiều các cách định nghĩa khác.

File: chapter_linear-networks/linear-regression-scratch_vn.md [hunk 11]
```diff
@@ -546,7 +552,7 @@ với dấu `@` ở đầu. Ví dụ: @aivivn.
 *
 
 <!-- Phần 6 -->
-*
+* Nguyễn Lê Quang Nhật
```
Comment (quangnhat185) [inline chapter_linear-networks/linear-regression-scratch_vn.md]:
```suggestion
* Nguyễn Lê Quang Nhật
* @tanduong 
* Vũ Hữu Tiệp
```

File: chapter_linear-networks/linear-regression-scratch_vn.md [hunk 12]
```diff
@@ -296,7 +299,7 @@ def squared_loss(y_hat, y):
 ## Defining the Optimization Algorithm
 -->
 
-## *dịch tiêu đề phía trên*
+## Định nghĩa thuật toán tối ưu
```
Comment (duythanhvn) [inline chapter_linear-networks/linear-regression-scratch_vn.md]:
```suggestion
## Định nghĩa Thuật toán Tối ưu
```
